pkg/config/validation/virtualservice.go

```diff
@@ -141,6 +141,13 @@ func validateHTTPRouteMatchRequest(http *networking.HTTPRoute, routeType HTTPRou
 					if header == nil {
 						errs = appendErrors(errs, fmt.Errorf("header match %v cannot be null", name))
 					}
+
+					if _, ok := header.GetMatchType().(*networking.StringMatch_Prefix); ok {
```

Comment:
Do the other two match need to check as well?

Reply:
Empty value for exact matcher is valid, and the regex matcher already fails to apply:

```
for: "STDIN": error when patching "STDIN": admission webhook "validation.istio.io" denied the request: configuration is invalid: "headers": regex string match should not be empty
```